distributor/distributor.go

```diff
@@ -441,61 +434,84 @@ func (d *Distributor) Query(ctx context.Context, from, to model.Time, matchers .
 			return err
 		}
 
-		if len(ingesters) < d.cfg.MinReadSuccesses {
-			return fmt.Errorf("could only find %d ingesters for query. Need at least %d", len(ingesters), d.cfg.MinReadSuccesses)
+		// We need a response from a quorum of ingesters, which is n/2 + 1.
+		minSuccess := (len(ingesters) / 2) + 1
```

Comment:
By my understanding of the DynamoDB paper, the constraint isn't that quorum is `n/2 + 1`, but rather that `r + w > n`, where `r` is minimum read quorum and `w` minimum write.